pkg/operator/ceph/object/controller.go

```diff
@@ -319,3 +329,32 @@ func updateStatus(client client.Client, name types.NamespacedName, status string
 	}
 	logger.Debugf("object store %q status updated to %q", name, status)
 }
+
+func (r *ReconcileCephObjectStore) handleDelete(objectstore *cephv1.CephObjectStore) error {
+	//  Wait for all the object buckets to be cleaned up
+	err := r.waitForObjectBucketCleanup(objectstore)
+	if err != nil {
+		return errors.Wrap(err, "failed to cleanup buckets")
+	}
+	return nil
+}
+
+func (r *ReconcileCephObjectStore) waitForObjectBucketCleanup(objectstore *cephv1.CephObjectStore) error {
+	selector := fmt.Sprintf("bucket-provisioner=%s", bucketProvisioner)
```

Comment:
You need the namespace to build it, just like in https://github.com/rook/rook/blob/master/pkg/operator/ceph/object/bucket/util.go#L60-L67